### DISCUSSION TITLE
2.7.2 build fails on MacOS 10.15.7 with homebrew, unsupported option '--with-readline-dir

### DISCUSSION THREAD

**Discussion by jrochkind** [rbenv/ruby-build#1862] [Build failures]
Trying to run: 

   ruby-build 2.7.2 ~/.rubies/ruby-2.7.2

I think it's actually failing trying to build the openssl dependency, `Installing openssl-1.1.1i...`.

Last lines of log file:

```
clang  -I. -Iinclude -fPIC -arch x86_64 -O3 -Wall --with-readline-dir=/usr/local/Cellar/readline/7.3.8 -DL_ENDIAN -DOPENSSL_PIC -DOPENSSL_CPUID_OBJ -DOPENSSL_IA32_SSE2 -DOPENSSL_BN_ASM_MONT -DOPENSSL_BN_ASM_MONT5 -DOPENSSL_BN_ASM_GF2m -DSHA1_ASM -DSHA256_ASM -DSHA512_ASM -DKECCAK1600_ASM -DRC4_ASM -DMD5_ASM -DAESNI_ASM -DVPAES_ASM -DGHASH_ASM -DECP_NISTZ256_ASM -DX25519_ASM -DPOLY1305_ASM -DOPENSSLDIR="\"/Users/jrochkind/.rubies/ruby-2.7.2/openssl/ssl\"" -DENGINESDIR="\"/Users/jrochkind/.rubies/ruby-2.7.2/openssl/lib/engines-1.1\"" -D_REENTRANT -DZLIB -DZLIB_SHARED -DNDEBUG -I/Users/jrochkind/.rubies/ruby-2.7.2/include  -MMD -MF apps/app_rand.d.tmp -MT apps/app_rand.o -c -o apps/app_rand.o apps/app_rand.c
clang: error: unsupported option '--with-readline-dir=/usr/local/Cellar/readline/7.3.8'
make[1]: *** [apps/app_rand.o] Error 1
make: *** [all] Error 2
```

It's trying to use a `--with-readline-dir` option that the `clang` it's trying to invoke does not recognize, I guess?

I have installed rubies before on this machine that I think now are failing -- maybe it now tries to build against openssl 1.1.1i when before it used previous openssl versions, and that's what's somehow failing? Not really sure.

**Comment by mislav** [comment]
Try `brew install openssl@1.1` and then building all Ruby versions by setting

```sh
RUBY_CONFIGURE_OPTS="--with-openssl-dir=$(brew --prefix openssl@1.1)" ruby-build 2.7.2 ...
```

That will avoid rebuilding openssl in favor of Homebrew's openssl.
